test/js_test.go

```diff
@@ -5362,17 +5362,6 @@ func testJetStreamMirror_Source(t *testing.T, nodes ...*jsServer) {
 		}
 	})
 
-	t.Run("bind to stream with wrong subject fails", func(t *testing.T) {
-		_, err := js.SubscribeSync("secret", nats.BindStream("origin"))
```

Comment:
btw why was this one deleted? Is it due to upgrading nats-server version?

Reply:
Yes, this PR is the companion to the new nats-server changes which introduce subject transforms to streams and as part of those changes the checking of the consumer filter having to match the stream's subject listeners has been removed, hence this test is no longer valid. It's been discussed at length between myself and Derek and myself and Tomasz both in the PR comments and in the #eng thread. TLDR is that the old assumption that all the messages in a stream are with subjects that match the current subject listen of the stream is no longer valid (actually stopped being so when we started allowing updating the subjects of an existing stream) and not only is it no longer valid but it's that artificial restriction is already a problem (i.e. people getting confused of how they can create a consumer and subscribe to a stream that doesn't listen to any Core NATS subjects), and would not allow source chaining (which is very much a desired functionality, especially once you have the subject transform functionality in the streams).

Reply:
Thanks for the update, in that case I would prefer we add something like the following replacing this test to clarify the behavior: 

```go
	t.Run("bind to stream with subject not in stream", func(t *testing.T) {
		// The Stream does not have a subject called 'nothing' but client is still able
		// to bind to the origin stream even though it cannot consume messages.
		// After updating the stream with the subject this consumer will be able to
		// match and receive messages.
		sub, err := js.SubscribeSync("nothing", nats.BindStream("origin"))
		if err != nil {
			t.Fatal(err)
		}
		_, err = sub.NextMsg(1 * time.Second)
		if !errors.Is(err, nats.ErrTimeout) {
			t.Fatal("Expected timeout error")
		}

		info, err := sub.ConsumerInfo()
		if err != nil {
			t.Fatal(err)
		}
		got := info.Stream
		expected := "origin"
		if got != expected {
			t.Fatalf("Expected %v, got %v", expected, got)
		}

		got = info.Config.FilterSubject
		expected = "nothing"
		if got != expected {
			t.Fatalf("Expected %v, got %v", expected, got)
		}

		t.Run("can consume after stream update", func(t *testing.T) {
			_, err = js.UpdateStream(&nats.StreamConfig{
				Name: "origin",
				Placement: &nats.Placement{
					Tags: []string{"NODE_0"},
				},
				Storage:  nats.MemoryStorage,
				Replicas: 1,
				Subjects: []string{"origin", "nothing"},
			})
			js.Publish("nothing", []byte("hello world"))

			msg, err := sub.NextMsg(1 * time.Second)
			if err != nil {
				t.Error(err)
			}
			got = msg.Subject
			expected = "nothing"
			if got != expected {
				t.Fatalf("Expected %v, got %v", expected, got)
			}

		})
	})
```

Reply:
(I will send a patch for this one after merging your PR which have just LGTMd)